app/chat/[id].tsx

```
@@ -81,6 +83,33 @@ export default function Chat() {
+                    tracking.event('chat_create_user_goal');
+                    return formatGoalForAI(goal);
+                }
+                case 'update-user-goal': {
+                    const args = toolCall.args as typeof tools['update-user-goal']['parameters']['_type'];
+                    const updatedGoal = await updateGoalAndSave(args.id, args, goals);
+                    if (!updatedGoal) {
+                        return `Goal #${args.id} not found. Max ID: ${goals.length}.`;
+                    } else {
+                        setGoals(updatedGoal);
+                        tracking.event('chat_update_user_goal');
+                        return formatGoalForAI(updatedGoal.find(g => g.id === args.id)!);
```

Biome (1.9.4)

[error] Forbidden non-null assertion.



(lint/style/noNonNullAssertion)